Header mobile

HelChris opened the pull request:
- add Zustand to json
- add header with logo
- add placeholder-img to assets

HelChris committed:
style: add placeholder image

HelChris committed:
feat: add logo to header

HelChris committed:
fix: add zustand

HelChris committed:
fix: add zustand

HelChris committed:
style: add justify-center

HelChris committed:
style: add to right side

HelChris commented:
- moved theme-toggle to right side of the logo
- added correct placement to logo

HelChris committed:
style: return header before themeToggle

Anclagen commented on src/App.jsx:
```diff
@@ -27,10 +27,10 @@ function App() {
+      <Header />
+      <div className="fixed top-4 right-4 z-50">
         <ThemeToggle />
       </div>
```
We should move the theme toggle into the header component now so the logo can be aligned with it and doesn't clip it on mobile.

Anclagen commented on src/components/ui/Header.jsx:
```diff
@@ -1,7 +1,11 @@
   return (
     <header>
-      <nav>Header</nav>
+      <div className="flex justify-center">
```
The designs show it aligned to the left, removing justify-center should achieve that.
```suggestion
      <div className="flex">
```

HelChris committed:
Merge branch 'dev' of https://github.com/Juniors-Dev/Juniors.dev into header-mobile

HelChris committed:
fix: merge language and light/dark mode into the header-mobile branch

HelChris commented:
Will add a navigation in header later, when that is needed. Can change the logo to SVG files when both light- and darkmode logo SVG is ready from the design team.

HelChris committed:
style: align to the left

Anclagen commented on src/components/ui/Header.jsx:
```diff
@@ -1,7 +1,11 @@
     <header>
-      <nav>Header</nav>
+      <div className="flex items-start">
+        <h5 className="text-[var(--h5-size)]">
```
Rather than using this approach we can define the text sizes for tailwind see [here](https://tailwindcss.com/docs/font-size#customizing-your-theme). 
```suggestion
        <h5 className="text-h5">
```

So we could add something like this to the @themes in the css and make use of tailwind for this (adjust as needed)
```css
  --text-h1: 3rem;           /* 48px */
  --text-h1--line-height: 3.5rem;
  --text-h1--font-weight: 700;

  --text-h2: 2rem;           /* 32px */
  --text-h2--line-height: 2.5rem;
  --text-h2--font-weight: 700;

  --text-h3: 1.5rem;         /* 24px */
  --text-h3--line-height: 2rem;
  --text-h3--font-weight: 600;

  --text-h4: 1.3rem;         /* 20.8px */
  --text-h4--line-height: 1.75rem;
  --text-h4--font-weight: 600;

  --text-h5: 1.1rem;         /* 17.6px */
  --text-h5--line-height: 1.5rem;
  --text-h5--font-weight: 500;

  --text-body: 1rem;         /* 16px */
  --text-body--line-height: 1.5rem;
  --text-body--font-weight: 400;

  --text-caption: 0.75rem;   /* 12px */
  --text-caption--line-height: 1rem;
  --text-caption--font-weight: 400;
 ```

HelChris committed:
style: add xs (500px) breakpoint

HelChris committed:
refactor: include themetoggle and languagetoggle in header. Add grid to header

HelChris committed:
refactor: move theme and language toggle into header

HelChris committed:
Merge branch 'dev' of https://github.com/Juniors-Dev/Juniors.dev into header-mobile

HelChris committed:
refactor: responsive w/xs breakpoint

HelChris committed:
refactor: add xs breakpoint

HelChris committed:
style: add SVG light and dark juniors logo

HelChris committed:
style: properly handle logo light/dark SVG

HelChris commented:
UPDATED:
- added logo SVG's to properly handle correct logo in light/dark
- added a grid to handle propotions/positioning according to design (mobile/tablet/desktop view)
- moved both theme toggle and language toggle into the header

How does it look?

Anclagen reviewed (dismissed):
Looks good, merging now. Only thing that we need to change is the toggle sizes so they match.

Anclagen committed:
Merge branch 'dev' of https://github.com/Juniors-Dev/Juniors.dev into header-mobile
